src/languages/php.js

```diff
@@ -18,18 +22,43 @@ export default function(hljs) {
       { begin: /\?>/ } // end php tag
     ]
   };
+  var SUBST = {
+    className: 'subst',
+    variants: [
+      { begin: /\$\w+/ },
+      { begin: /\{\$/, end: /\}/ }
+    ]
+  };
+  var SINGLE_QUOTED = hljs.inherit(hljs.APOS_STRING_MODE, {
+    illegal: null,
+  });
+  var DOUBLE_QUOTED = hljs.inherit(hljs.QUOTE_STRING_MODE, {
+    illegal: null,
+    contains: hljs.QUOTE_STRING_MODE.contains.concat(SUBST),
+  });
+  var HEREDOC = {
+    begin: /<<<[ \t]*(?<marker>\w+)\n(?:.*\n)*?^[ \t]*\k<marker>\b/,
```

Comment:
Did we not discuss simplifying all this?

Reply:
yes i moved the "<<<" into the end_same_as_begin, it seems good now, the regex is clear with the named group marker, ready yo be merged ?

Reply:
My point is why do we need lines 40/41 at all? ALL we really need here is just the END_SAME_AS_BEGIN rule without any other rules, no?

Reply:
right, I missed that copying from ruby.js, fixed!